README.md

```diff
@@ -144,33 +146,75 @@ $ make -C build.assets build-binaries
 
 ### Local Build
 
-To perform a build on your host, ensure you have installed Go. In order to
-include the Rust-powered features like Desktop Access, you'll
-also need `cargo` and `rustc`. The current versions of these tools can be found
-in `build.assets/Makefile`.
+**Important:**
 
-```bash
-# get the source & build:
+* The Go compiler is somewhat sensitive to the amount of memory: you will need
+  **at least** 1GB of virtual memory to compile Teleport. A 512MB instance
+  without swap will **not** work.
+* This will build the latest version of Teleport, **regardless** of whether it
+  is stable. If you want to build the latest stable release, run `git checkout`
+  to the corresponding tag (for example, run `git checkout v8.0.0`) **before**
+  performing a build.
+* `tsh` binaries with Touch ID support are only functional using binaries signed
+  with Teleport's Apple Developer ID and notarized by Apple. If you are a Teleport
+  maintainer, ask the team for access.
+
+#### Dependencies
+
+Ensure you have installed correct versions of necessary dependencies:
+* `Go` version from
+  [go.mod](https://github.com/gravitational/teleport/blob/master/go.mod#L3)
+* If you wish to build the Rust-powered features like Desktop Access, `Rust` and `Cargo` version from
+  [build.assets/Makefile](https://github.com/gravitational/teleport/blob/master/build.assets/Makefile#L21)
+  (search for `RUST_VERSION`)
+* For `tsh` version > `10.x` with FIDO support, you will need `libfido` and `openssl 1.1` installed locally
+
+For an example of Dev Environment setup on a Mac, see [these instructions](README_local_MacOS.md). 
+
+#### Perform a build
+
+Get the source
+
+```shell
 $ git clone https://github.com/gravitational/teleport.git
 $ cd teleport
+```
+
+To perform a build 
+
+```shell
 $ make full
+```
 
-# create the default data directory before starting:
-$ sudo mkdir -p -m0700 /var/lib/teleport
-$ sudo chown $USER /var/lib/teleport
+To build `tsh` with Apple TouchID support enabled:
```

Comment:
My 2c: I would ditch this or just mention the build tag in the "Important" section above. I'm afraid that, if we leave this here, folks will try to build with Touch ID support and not see/ignore the warning.

Reply:
Yeah, you are right. I'll see if we can consolidate these instructions and the note about TouchID/Apple developer thing.

Reply:
Moved the section around in https://github.com/gravitational/teleport/pull/15092/commits/a33951a6a363735adf3ef7a6058f30344c6122a3